docs/hooks.py

```diff
@@ -1,4 +1,10 @@
-import os
+import subprocess
+from pathlib import Path
 
 def on_pre_build(**kwargs):
-    os.system('nix run .#treefmt -- --help > ./snippets/usage.txt')
\ No newline at end of file
+    with Path("./snippets/usage.txt").open("w") as f:
+        subprocess.run(
+            ["nix", "run", ".#treefmt", "--", "--help"],
+            stdout=f,
+            check=True,
+        )
```

Comment:
A much simpler alternative:

```suggestion
    assert 0 == os.system('nix run .#treefmt -- --help > ./snippets/usage.txt')
```

I personally like to avoid bash/sh/whatever if I can, though. One less dependency.

Reply:
I changed this to executing the `treefmt` binary directly, as this causes issues in the `docs` package https://github.com/numtide/treefmt/pull/551/commits/583681d0794690c2544bf65db057245e50a130fb